crates/sui-storage/src/write_path_pending_tx_log.rs

```diff
@@ -18,6 +19,7 @@ use typed_store::DBMapUtils;
 use typed_store::{rocks::DBMap, traits::Map};
 
 pub type IsFirstRecord = bool;
+const NUM_SHARDS: usize = 4096;
```

Comment:
nit: We probably don't need that many shard for a simple operation like check-insert, something like 128 would be more then enough.